### PR TITLE
[Merged by Bors] - feat(Data/Matrix/Notation): lemmas about numeric literals and small matrices

### PR DESCRIPTION
**Pull request by eric-wieser**
[Zulip](https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/Complex.20Numbers.20in.20Matrices/near/421653137)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Matrix/Notation.lean [hunk 1]
```diff
@@ -437,6 +437,32 @@ theorem one_fin_three : (1 : Matrix (Fin 3) (Fin 3) α) = !![1, 0, 0; 0, 1, 0; 0
...
+theorem natCast_fin_three (n : ℕ) :
+    (n : Matrix (Fin 3) (Fin 3) α) = !![↑n, 0, 0; 0, ↑n, 0; 0, 0, ↑n] := by
+  ext i j
+  fin_cases i <;> fin_cases j <;> rfl
+
+-- See note [no_index around OfNat.ofNat]
+theorem ofNat_fin_two (n : ℕ) [n.AtLeastTwo] :
+    (no_index (OfNat.ofNat n) : Matrix (Fin 2) (Fin 2) α) =
+      !![OfNat.ofNat n, 0; 0, OfNat.ofNat n] :=
+  natCast_fin_two n
```
Comment (adomani) [inline Mathlib/Data/Matrix/Notation.lean:456]:
```suggestion
  natCast_fin_two _
```
very minor nit-pick, for consistency with the proof of `ofNat_fin_three`.  ;)

**Comment by adomani** [comment]
I hope that it is on to

maintainer merge

a PR that has not passed CI yet.

**Comment by adomani** [comment]
I hope that it is on to

maintainer merge

a PR that has not passed CI yet.

**Comment by adomani** [comment]
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**